sympy/assumptions/cnf.py

```diff
@@ -143,7 +143,27 @@ def to_NNF(expr):
     """
     Generates the Negation Normal Form of any boolean expression in terms
     of AND, OR, and Literal objects.
+
+    Examples
+    ========
+
+    >>> from sympy import Q
+    >>> from sympy.assumptions.cnf import to_NNF
+    >>> from sympy.abc import x
+    >>> expr = Q.even(x) & ~Q.positive(x)
+    >>> to_NNF(expr)
+    (Literal(Q.even(x), False) & Literal(Q.positive(x), True))
+
+    ``to_NNF`` decomposes the predicate into a combination of primitive
+    predicates if possible.
```

Comment:
It doesn't seem like this belongs as part of the `to_NNF` function.

Reply:
Do you mean that predicate decomposition should be done by some other function?

Reply:
It just seems messy to me to do it as part of this function which has a clearly defined purpose: take a logical expression and convert it to NNF (without applying any interpretation to the meaning of the literals).

Reply:
What about giving keyword argument `decompose=True/False` to this function?

Reply:
Perhaps it could take a `composite_map` argument or something like that.

Reply:
Done.